keps/sig-architecture/20190731-production-readiness-review-process.md

```diff
@@ -0,0 +1,258 @@
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+This proposal introduces a production readiness review process for features
+merging as alpha or graduating to beta or GA. The review process is intended to
+ensure that features merging into Kubernetes are supportable, can be safely
+operated in production environments, and can be disabled or rolled back in the
+event they cause increased failures in production.
+
+## Motivation
+
+Kubernetes has grown quickly and organically. The KEP process introduced a
+mechanism to help build consensus and ensure the community supports the
+direction and implementation of various features. It provides a way to document
+release and graduation criteria, but it leaves that criteria up to the KEP
+authors and approvers to define. Because these are normally the developers and
+the SIGs associated with the KEP, there is not always a clear representation of
+the view of cluster operators. This can result in operational and supportability
+oversights in the graduation criteria.
+
+This KEP proposes a process to ensure that production concerns are addressed in
+all new features, at a level appropriate to the features' maturity levels.
+
+* TBD: Document some experiences that motivate the need for this.
+
+### Goals
+
+* Define production readiness criteria for alpha, beta, and GA features.
+* Define a production readiness review gate and process for all features.
+* Utilize existing tooling with prow to enforce the process.
+
+### Non-Goals
+
+* Building new tooling to enforce the process.
+
+## Proposal
+
+* Document production readiness criteria in a process document in the
+  kubernetes/community repository. Different levels of readiness may be
+  specified for different feature maturity levels.
+
+* Develop a production readiness questionnaire to ensure that the feature
+  authors consider and document operational aspects of the feature. Example
+  items:
+  * Feature enablement
+    - How can this feature be disabled in a live cluster? Will that require
+      downtime for the control plane?
```

Comment:
Can this feature be enabled / disabled on an existing node or must the node be reprovisioned in response?

Reply:
Thanks, I added this and @vishh's below. But please realize these are examples - we will develop the actual questionnaire completely as part of the "implementation" of this KEP. Still, happy to put in anything here to get that started (maybe we'll end up with most of what we need already).